migrate-project-memberships

From: Bryan Larsen <bryan@larsen.st>

Run the migration generator to have the required foreign keys added to
the database:

    $ ./script/generate hobo_migration
---

 ...201135345_hobo_migration_project_memberships.rb |   16 ++++++++++++++++
 db/schema.rb                                       |   12 +++++++++++-
 2 files changed, 27 insertions(+), 1 deletions(-)
 create mode 100644 db/migrate/20091201135345_hobo_migration_project_memberships.rb


diff --git a/db/migrate/20091201135345_hobo_migration_project_memberships.rb b/db/migrate/20091201135345_hobo_migration_project_memberships.rb
new file mode 100644
index 0000000..e6c7e1f
--- /dev/null
+++ b/db/migrate/20091201135345_hobo_migration_project_memberships.rb
@@ -0,0 +1,16 @@
+class HoboMigrationProjectMemberships < ActiveRecord::Migration
+  def self.up
+    create_table :project_memberships do |t|
+      t.datetime :created_at
+      t.datetime :updated_at
+      t.integer  :project_id
+      t.integer  :user_id
+    end
+    add_index :project_memberships, [:project_id]
+    add_index :project_memberships, [:user_id]
+  end
+
+  def self.down
+    drop_table :project_memberships
+  end
+end
diff --git a/db/schema.rb b/db/schema.rb
index c434f2d..1a958c2 100644
--- a/db/schema.rb
+++ b/db/schema.rb
@@ -9,7 +9,17 @@
 #
 # It's strongly recommended to check this file into your version control system.
 
-ActiveRecord::Schema.define(:version => 20091201135235) do
+ActiveRecord::Schema.define(:version => 20091201135345) do
+
+  create_table "project_memberships", :force => true do |t|
+    t.datetime "created_at"
+    t.datetime "updated_at"
+    t.integer  "project_id"
+    t.integer  "user_id"
+  end
+
+  add_index "project_memberships", ["project_id"], :name => "index_project_memberships_on_project_id"
+  add_index "project_memberships", ["user_id"], :name => "index_project_memberships_on_user_id"
 
   create_table "projects", :force => true do |t|
     t.string   "name"
